apps/greencheck/importers/importer_interface.py

```diff
@@ -17,67 +19,138 @@ def fetch_data_from_source(cls) -> list:
+        with transaction.atomic():
+            # Retrieve all ASN entries assosiated with this hosting provider
+            hosting_provider = Hostingprovider.objects.get(pk=cls.hosting_provider_id)
+            entries = GreencheckASN.objects.select_for_update().filter(
+                hostingprovider=hosting_provider
+            )
 
-        gc_ip, created = GreencheckIp.objects.update_or_create(
-            active=True, ip_start=network[1], ip_end=network[-1], hostingprovider=hoster
-        )
-        gc_ip.save()  # Save the newly created or updated object
+            # Iterate database entries
+            for entry in entries:
+                if entry.asn in active_networks:
+                    active_networks.pop(active_networks.index(entry.asn))
+                    entry.active = True
+                    entry.save()
+
+                    updated_networks += 1
+                    logger.debug(entry)
+                elif entry.active == True:
+                    entry.active = False
+                    entry.save()
+
+                    updated_networks += 1
+                    logger.debug(entry)
+
+            # Iterate remaining list items (that are not in the database)
+            if active_networks:
+                for asn_network in active_networks:
+                    entry, created = GreencheckASN.objects.update_or_create(
+                        active=True, asn=asn_network, hostingprovider=hosting_provider
+                    )
+                    entry.save()
+
+                    if created:
+                        logger.debug(entry)
+
+        return updated_networks
+
+    def update_ip(cls, active_networks: List[str]) -> int:
+        updated_networks = 0
+
+        # Convert networks (xxx.xxx.xxx.xxx/yy) to a
+        # start and end ip address
+        tmp_list = []
+        for ip_network in active_networks:
+            network = ipaddress.ip_network(ip_network)
+            tmp_list.append((str(network[1]), str(network[-1])))
+        active_networks = list(set(tmp_list))
+
+        logger.debug("Running atomic database transaction. Altered IP ranges:")
+        with transaction.atomic():
+            # Retrieve all IP(v4 and v6) entries assosiated to this hosting provider
+            hosting_provider = Hostingprovider.objects.get(pk=cls.hosting_provider_id)
+            entries = GreencheckIp.objects.select_for_update().filter(
+                hostingprovider=hosting_provider
+            )
+
+            # Iterate database entries
+            for entry in entries:
+                if (entry.ip_start, entry.ip_end) in active_networks:
+                    active_networks.pop(
+                        active_networks.index((entry.ip_start, entry.ip_end))
+                    )
+                    entry.active = True
+                    entry.save()
+
+                    updated_networks += 1
+                    logger.debug(entry)
+                elif entry.active == True:
```

Comment:
Can you add a one liner comment about why we make things as true or false, or add it at the top of the loop explaining the intent here?